### ISSUE TITLE
Custom context provider is not showing up in version 0.8.45

### ISSUE THREAD

**Issue by tranquochuy645** [continuedev/continue#1936]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

- OS: Windows 10
- Continue: 0.8.45
- IDE: VScode
- config.json:
  
```json
{
  "defaultModel": {
    "title": "local ",
    "apiBase": "http://127.0.0.1:11434",
    "provider": "ollama",
    "model": "llama3.1-8b"
  },
  "models": [
    {
      "title": "local ",
      "apiBase": "http://127.0.0.1:11434",
      "provider": "ollama",
      "model": "llama3-8b"
    }
  ],
  "customCommands": [
    {
      "name": "test",
      "prompt": "{{{ input }}}\n\nWrite a comprehensive set of unit tests for the selected code. It should setup, run tests that check for correctness including important edge cases, and teardown. Ensure that the tests are complete and sophisticated. Give the tests just as chat output, don't edit any file.",
      "description": "Write unit tests for highlighted code"
    }
  ],
  "tabAutocompleteModel": {
    "title": "Starcoder 3b",
    "provider": "ollama",
    "model": "starcoder2:3b",
    "apiBase": "http://127.0.0.1:11434"
  },
  "allowAnonymousTelemetry": false,
  "embeddingsProvider": {
    "provider": "ollama",
    "model": "nomic-embed-text",
    "apiBase": "http://127.0.0.1:11434"
  },
  "docs": []
}
``` 

-config.ts:

```typescript
const getSomePrivateData = async (query: string, extras: ContextProviderExtras): Promise<ContextItem[]> => {
  return [
    {
      name: "name",
      description: "some des",
      content:"Hi mom"
    }
  ]
}

const RagContextProvider: CustomContextProvider = {
  title: "Private data",
  getContextItems: getSomePrivateData,
};

export function modifyConfig(config: Config): Config {
  if (!config.contextProviders) {
    config.contextProviders = [];
  }
  config.contextProviders.push(RagContextProvider);
  return config;
}
```

### Description

This bug happened after I updated Continue to version 0.8.45. I didn't find any change mentioned in v0.8.45 about the custom context provider, so I believe this is a bug:

-Custom context provider is not showing up in the dropdown menu "@" in version 0.8.45
![image](https://github.com/user-attachments/assets/27ef5dc1-ee70-4096-b3f1-dda4151690bb)

-But it works in version  0.8.43 with the same setup
![image](https://github.com/user-attachments/assets/c8d0ea48-d880-460e-a927-e65fc5ce0046)

**Comment by Patrick-Erichsen** [comment]
Hi @tranquochuy645 , thanks for opening this issue.

I was able to confirm the issue on both `v0.9.194` and `v0.8.45`.

However, running off the latest commit in `dev`, I'm able to see the custom context provider.  So, I'm not sure what the root of the issue is here but I believe it's been resolved.

We will be publishing a new pre-release (`v0.9.195`) in the next day or two. If you don't mind, install `v0.9.195` and report back whether or not you're still having issues.

**Comment by tranquochuy645** [comment]
Hi @Patrick-Erichsen , thank you for the effort. 

Unfortunately, switching to v0.9.195 didn't solve the issue for me.
It looks like many people are also facing this bug #1947

**Comment by Patrick-Erichsen** [comment]
You are correct, thanks for circling back. The issue appears to be with our node config in our build system vs local development. Hoping to have a fix by tomorrow!

**Comment by Patrick-Erichsen** [comment]
@tranquochuy645 - this should be resolved with https://github.com/continuedev/continue/pull/1975.

Previously we were targeting CommonJS in the core package, which transpiled the dynamic import used in our config loading, but we recently moved to targeting ESNext and now need to explicitly transpile dynamic imports when packaging the extension.

Apologies for the bug, and thanks all for your help troubleshooting!

This fix will be included in the next pre-release later today.

**Comment by tranquochuy645** [comment]
@Patrick-Erichsen , 

Thank you so much, I confirm it is fixed for me in v0.8.46
